nixos/modules/services/logging/vector.nix

```diff
@@ -0,0 +1,73 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+let
+  cfg = config.services.vector;
+
+in {
+  options.services.vector = {
+    enable = mkOption {
+      type = types.bool;
+      default = false;
+      description = ''
+        Enable the vector daemon.
+      '';
+    };
+
+    package = mkOption {
+      type = types.package;
+      default = pkgs.vector;
+      defaultText = "pkgs.vector";
+      description = ''
+        The vector package that should be used.
+      '';
+    };
+
+    user = mkOption {
+      type = types.str;
+      default = "vector";
+      description = ''
+        User under which the vector service runs.
+      '';
+    };
+
+    group = mkOption {
+      type = types.str;
+      default = "vector";
+      description = ''
+        Group under which the vector service runs.
+      '';
+    };
+
+    configuration = mkOption {
+      type = (pkgs.formats.json {}).type;
+      default = {};
+      description = ''
+        Specify the configuration for Vector in Nix.
+      '';
+    };
+  };
+
+  config = mkIf cfg.enable {
+
+    users.groups.${cfg.group} = {};
+    users.users.${cfg.user} = {
+      description = "Vector service user";
+      group = cfg.group;
+      isSystemUser = true;
+    };
+    systemd.services.vector = {
```

Comment:
You should consider using the [unit provided by upstream](https://github.com/timberio/vector/blob/0f63b7efa02325d97a18ba5ed548c6ded8004131/distribution/systemd/vector.service) instead.